models/loss_function.py

```diff
@@ -4,80 +4,53 @@
+def l1_loss(y_true, y_pred, config):
     """ l1 loss
             y_true: true targets tensor.
             y_pred: predictions tensor.
     """
 
-    sum_loss=0
+    sum_loss = 0
     for class_index in range(config['channel_label_num']):
-        y_t=y_true[...,class_index]
-        y_p=y_pred[...,class_index]
-        sum_loss+=K.mean((tf.abs(y_p - y_t)))*config['loss_channel_weight'][class_index]
+        y_t = y_true[..., class_index]
+        y_p = y_pred[..., class_index]
+        sum_loss += K.mean((tf.abs(y_p - y_t))) * config['loss_channel_weight'][class_index]
     return sum_loss
 
-def l2_loss(y_true, y_pred,config):
-    sum_loss=0
+
+def l2_loss(y_true, y_pred, config):
+    sum_loss = 0
     for class_index in range(config['channel_label_num']):
-        y_t=y_true[...,class_index]
-        y_p=y_pred[...,class_index]
-        sum_loss+=K.mean(K.pow(y_p - y_t,2))*config['loss_channel_weight'][class_index]
+        y_t = y_true[..., class_index]
+        y_p = y_pred[..., class_index]
+        sum_loss += K.mean(K.pow(y_p - y_t, 2)) * config['loss_channel_weight'][class_index]
     return sum_loss
 
 
-def dice_loss(y_true, y_pred,config):
+def dice_loss(y_true, y_pred, config):
     """ Dice loss
             y_true: true targets tensor.
             y_pred: predictions tensor.
             Dice calculation with smoothing to avoid division by zero
+
+
+
     """
-    #smooth = 1E-16
-    smooth = K.epsilon()
-    sum_loss,weight_sum=0,0
-    for class_index in range(config['channel_label_num']):
-        y_t=y_true[...,class_index]
-        y_p=y_pred[...,class_index]
-        intersection = K.sum(K.abs(y_t * y_p), axis=-1)
-        loss =1 - (2. * intersection + smooth) / (K.sum(K.square(y_t), -1) + K.sum(K.square(y_p), -1) + smooth)
-        sum_loss+=loss*config['loss_channel_weight'][class_index]
-        weight_sum+=config['loss_channel_weight'][class_index]
-    return sum_loss/(weight_sum+smooth)
-
-def dice_coefficient_loss(y_true, y_pred,config, smooth=K.epsilon(), axis=None):
-    """ Dice coefficient along specific axis (same as  1+dice_loss() if axis=None)
-            y_true: true targets tensor.
-            y_pred: predictions tensor.
-            smooth: smoothing parameter to avoid division by zero
-            axis: along which to calculate Dice
-    """
-    intersection = K.sum(K.abs(y_true * y_pred), axis=axis)
-    return -(2. * intersection + smooth) / (K.sum(K.abs(y_true), axis=axis) + K.sum(K.abs(y_pred), axis=axis) + smooth)
 
-def dice_loss_v2(y_true, y_pred,config):
     smooth = 1E-16
+    # smooth = K.epsilon()
     sum_loss, weight_sum = 0, 0
     for class_index in range(config['channel_label_num']):
         y_t = y_true[..., class_index]
         y_p = y_pred[..., class_index]
-        intersection = K.sum(K.abs(y_t * y_p), axis=-1)
-        loss = 1 - (2. * intersection + smooth) / (K.sum(K.abs(y_t)) + K.sum(K.abs(y_p)) + smooth)
+        intersection = K.sum(K.abs(y_t * y_p))
```

Comment:
can't accept redefinition of dice loss -> keep compatibility. Is axis=-1 needed?

Reply:
The axis=-1 should be deleted.